### PR TITLE
[cmake] Add CMake presets configuration

### PR DESCRIPTION
**Pull request by idzm**
Add [CMake presets](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html) [configuration](https://github.com/savushkin-r-d/ptusa_main/blob/a422ab4887d80d9c1bd40393156dfbdc093a7f63/CMakePresets.json) to build binaries for different targets and controllers (Linux, Windows, **AXC F 2152** different versions).

### CODE REVIEW

File: PAC/linux/PAC_PLCnext/PtusaPLCnextEngineer/PtusaMainPrg.hpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    public:
+        // properties
+
+    public:
+        // operations
+        void Execute() override;
+
+        //#port
+        //#attributes(Output|Retain)
+        uint8 NVRAM[ 49000 ] = {0};
```
Comment (dimabeel) [inline PAC/linux/PAC_PLCnext/PtusaPLCnextEngineer/PtusaMainPrg.hpp:40]: Should define a constant var

File: PAC/linux/PAC_PLCnext/PtusaPLCnextEngineer/PtusaMainPrg.cpp [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+                           lua_gc( G_LUA_MANAGER->get_Lua(), LUA_GCCOUNTB, 0 ) );
+                       G_LOG->write_log( i_log::P_ALERT );
+                       }
+
+                   sprintf( G_LOG->msg,
+                       "Main control cycle performance : "
+                       "avg = %lu, max = %4u, tresh = %4u ms (%4u cycles, Lua mem = %d b).",
+                       avg_time, max_iteration_cycle_time, TRESH_AVG,
+                       cycles_per_period,
+                       lua_gc( G_LUA_MANAGER->get_Lua(), LUA_GCCOUNT, 0 ) * 1024 +
```
Comment (dimabeel) [inline PAC/linux/PAC_PLCnext/PtusaPLCnextEngineer/PtusaMainPrg.cpp:202]: Constant should be variable

File: PAC/linux/PAC_PLCnext/PtusaPLCnextEngineer/PtusaMainPrg.cpp [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+                   {
+                   u_long avg_time = all_time / cycles_cnt;
+
+                   if ( TRESH_AVG < avg_time )
+                       {
+                       sprintf( G_LOG->msg,
+                           "Main control cycle avg time above threshold : "
+                           "%4lu > %4u ms (Lua mem = %d b).",
+                           avg_time, TRESH_AVG,
+                           lua_gc( G_LUA_MANAGER->get_Lua(), LUA_GCCOUNT, 0 ) * 1024 +
```
Comment (dimabeel) [inline PAC/linux/PAC_PLCnext/PtusaPLCnextEngineer/PtusaMainPrg.cpp:192]: Constant var

File: PAC/linux/PAC_PLCnext/PtusaPLCnextEngineer/PtusaMainPrg.cpp [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+#ifdef TEST_SPEED
+               static u_long st_time;
+               static u_long all_time   = 0;
+               static u_long cycles_cnt = 0;
+
+               st_time = get_millisec();
+               cycles_cnt++;
+       #endif // TEST_SPEED
+
+            lua_gc(G_LUA_MANAGER->get_Lua(), LUA_GCSTEP, 200);
```
Comment (dimabeel) [inline PAC/linux/PAC_PLCnext/PtusaPLCnextEngineer/PtusaMainPrg.cpp:90]: Constant var

**Review by dimabeel — DISMISSED** [review]

**Review by dimabeel — APPROVED** [review]

**Review by rartem — APPROVED** [review]

**Pull request merged**